OpenRA.Mods.Common/Traits/Encyclopedia.cs

```diff
@@ -24,12 +24,10 @@ public class EncyclopediaInfo : TraitInfo
 		[Desc("Group under this heading.")]
 		public readonly string Category;
 
-		public override object Create(ActorInitializer init) { return Encyclopedia.Instance; }
+		public override object Create(ActorInitializer init) { return new Encyclopedia(); }
```

Comment:
This was probably intentional to not create a new instance for each `Encyclopedia` trait info (which we don't need). Can we just return `null` here?

Reply:
Gave that a go but the game wasn't happy dealing with a null.

Switching it to a struct stopped the rule from firing. The other option if we don't like that workaround is just to suppress the rule here which is firing in error.

Reply:
I'd also be ok with leaving this as is with a `TODO` or an open issue that we should introduce a new mechanism that supports having trait infos without concrete instances.

Reply:
IIRC we have such traits already

Reply:
From a quick look I found `MustBeDestroyed`, `EditorOnlyTooltipInfo`